### DISCUSSION TITLE
Reinjecting?

### DISCUSSION THREAD

**Discussion by JunkMeal** [violentmonkey/violentmonkey#2130] [Q&A]
Hey can i somehow reinject a script that I am working on locally which i am tracking through a http server? This would make development so much faster because after every change I have to wait for the website to load again

**Comment by gera2ld** [comment]
Check https://github.com/lisonge/vite-plugin-monkey

**Comment by JunkMeal** [comment]
for some reason when i use it, it just refreshes the page

**Reply by gera2ld** [reply]
Only side-effect-free components and functions can be hot-reloaded. Otherwise, a full reload is required because JavaScript cannot be undone once injected.

P.S. This relates to how JavaScript works and has nothing to do with userscripts or Violentmonkey.
